### DISCUSSION TITLE
Is there a built-in way to parse an application/octet-stream into bytes?

### DISCUSSION THREAD

**Discussion by najtin** [actix/actix-web#2804] [Q&A] [answered]
Hey, i want to upload a file. I found actix multipart and it is working fine, but i was wondering if there is a built-in way to parse an application/octet-stream into bytes? Can someone help me out? Thanks in advance for your time.

**Comment by robjtede** [comment, marked as answer]
`Bytes` extractor will get any request's body bytes.

Eg:
```rust
use actix_web::web::Bytes;

#[get("/")]
async fn handler(body: Bytes) -> ... {
   ...
}
```

You can check for the content type too if you want

Eg:
```rust
use actix_web::{http::header::ContentType, web::{Bytes, Header}};

#[get("/")]
async fn handler(body: Bytes, ct: Header<ContentType>) -> ... {
   let mime_type = ct.into_inner().0;
   ...
}
```
